src/sync/sync_session.cpp

```diff
@@ -455,8 +461,8 @@ std::function<void()> SyncSession::create_notifier_invocation(const NotifierPack
     // A notifier is expired if at least as many bytes have been transferred
     // as were originally considered transferrable.
     is_expired = !package.is_streaming && transferred >= package.captured_transferrable;
-    return [=](){
-        package.notifier(transferred, transferrable);
+    return [=, notifier=package.notifier](){
```

Comment:
`[=]` says to capture any referenced variables by copy, so the lambda stores a copy of the object `package` references. It doesn't matter what happens to `package` after the lambda is created.